src/hardware_manager.rs

```diff
@@ -0,0 +1,332 @@
+use crate::data_logger::DataLogger;
+use crate::server::protocols::v1::packages;
+use lazy_static::lazy_static;
+use paperclip::actix::Apiv2Schema;
+use serde::{Deserialize, Serialize};
+use std::convert::From;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread;
+use strum_macros::EnumIter;
+
+#[derive(Default)]
+struct NavigationManager {
+    navigator: navigator_rs::Navigator,
+    monitor: Option<std::thread::JoinHandle<()>>,
+    datalogger: Option<std::thread::JoinHandle<()>>,
+}
+
+#[derive(Debug, Clone, Default, Copy)]
+struct Data {
+    state: navigator_rs::SensorData,
+}
+
+macro_rules! with_navigator {
+    () => {
+        NavigationManager::get_instance().lock().unwrap().navigator
+    };
+}
+
+macro_rules! impl_from_enum {
+    ($from:ty, $to:ty, $($variant:ident),+ $(,)?) => {
+        impl From<$from> for $to {
+            fn from(item: $from) -> Self {
+                match item {
+                    $(
+                        <$from>::$variant => <$to>::$variant,
+                    )+
+                }
+            }
+        }
+    };
+}
+
+lazy_static! {
+    static ref NAVIGATOR: Arc<Mutex<NavigationManager>> = Default::default();
+}
+
+lazy_static! {
+    static ref DATA: Arc<RwLock<Data>> = Default::default();
+}
+
+impl NavigationManager {
+    pub fn get_instance() -> &'static Mutex<Self> {
+        &NAVIGATOR
+    }
+
+    pub fn init_monitor(refresh_interval: u64) {
+        NavigationManager::get_instance().lock().unwrap().monitor = Some(
+            thread::Builder::new()
+                .name("Navigator service - monitor".into())
```

Comment:
This name will not be distinguishable in `htop` or `ps`. Please keep the thread names under 15 characters. 

Since they are subprocesses of the Navigator binary, you can name it as just `monitor` for example.